// rd-react-overlay/examples/app.tsx
import * as React from "react";
import { OverlayContainer } from "../src/overlayContainer";
import { Overlay } from "../src/overlay";
import { Alignment, AlignType } from "../src/positioning";


export interface AppState {
    visible: boolean;
    alignment?: Alignment;
}

export class App extends React.Component<void, AppState> {

    constructor(props: any) {
        super(props);

        this.state = {
            visible: false,
            alignment: {
                target: {
                    horizontal: AlignType.Left,
                    vertical: AlignType.Bottom
                },
                overlay: {
                    horizontal: AlignType.Left,
                    vertical: AlignType.Top
                }
            }
        }
    }

    render() {
        return (
            <OverlayContainer>
                <div style={({width: "100px"})} onClick={() => this.setState({visible: true})}>
                    Lorem ipsum dolor sit amet, consectetur adipisicing elit. At ea expedita modi quis, ullam voluptatem?
                    Ab alias aperiam dolor doloribus, explicabo fugit, ipsa ipsum laboriosam laudantium, minima optio quod sint?
                </div>
                <Overlay visible={this.state.visible}
                         onClickOutside={(clickedOnContainer) => this.setState({visible: clickedOnContainer})}
                         alignment={this.state.alignment}>
                    {
                        (left, top) =>
                            <div style={({position: 'absolute', top: top, left: left})}>
                                Lorem ipsum dolor sit.
                            </div>
                    }
                </Overlay>
            </OverlayContainer>
        )
    }
}
